### ISSUE TITLE
Unique index on DateTime

### ISSUE THREAD

**Issue by ibox4real** [vlucas/spot2#115]
When i add unique index to DateTime field type i get this error when i use save method: 
<code>Catchable Fatal Error: Object of class DateTime could not be converted to string</code>

**Comment by vlucas** [comment]
Are you getting this in a `migrate`?

This can happen if you are setting a new `DateTime` object on the `default` property. The `default` property is used during a migrate to set the default value in the actual database. If you want a default `DateTime` object as the default value for new entities, use the `value` property instead. This does the same thing, but it does not translate to the database itself like `default` does.

**Comment by ibox4real** [comment]
No. I'm just building a new entitie <code>$entity = $mapper->build(['date'=> new \DateTime($timestamp)]</code>) and am trying to save it <code>$mapper->save($entity)</code>. While having unique key on that field it throws an exception.

**Comment by vlucas** [comment]
Ah, so it's in the validation. Can you provide a full stack trace so I can pinpoint where the error is happening please?

**Comment by sebastian-galle** [comment]
same issue here. Mapper::validate query conditions are unconverted php values
` if (!empty($uniqueWhere)) {
            foreach ($uniqueWhere as $field => $value) {
                if (!is_array($value)) {
                    $value = [$field => $entity->$field];
                }
                if (!in_array(null, $value, true) && $this->first($value) !== false) {`
